packages/toolpad-app/src/runtime/evalJsBindings.ts

```diff
@@ -239,11 +240,41 @@ export default function evalJsBindings(
   return mapValues(bindings, (binding, bindingId) => {
     const { scopePath } = binding;
 
+    let bindingResult = results[bindingId];
+
+    const mergeNestedBindings = (value: unknown, parentBindingId: string) => {
+      if (value && typeof value === 'object') {
+        for (const nestedPropName of Object.keys(value)) {
+          const nestedBindingId = `${parentBindingId}${
+            Array.isArray(value) ? `[${nestedPropName}]` : `.${nestedPropName}`
+          }`;
+
+          const nestedBindingResultValue = results[nestedBindingId]?.value;
+          if (nestedBindingResultValue) {
+            bindingResult = updatePath(
```

Comment:
I'd prefer this written with nested `update` calls as we do everywhere else. It's slightly more verbose but it's more type-safe, typescript will yell at us if we update the `AppDom` but forget to update this statement.
It's probably slightly more performant as well and it doesn't create clones if it doesn't need to.

We can always start using `immer` to simplify these sort of updates ina  type safe way.

Reply:
Got it, I'll try to use `update` for now.

Reply:
The reason I created and used an `updatePath` method is because lodash makes it easy to update an object by passing a string with the path of the property we want to change (it also needs `clone` to be immutable, though). And in this case we could have a unknown number of nested objects or arrays.

I could create an issue to write our own method for this later using our own `update` instead of lodash methods?